frontend/src/views/FuelCodes/AddFuelCode/AddFuelCode.jsx

```diff
@@ -58,284 +39,222 @@ const AddFuelCodeBase = () => {
-      params.data.isValid = false
-      params.data.validationMsg = `${field} ${message.response?.data?.detail[0]?.msg}`
-    }
-    setAlertMessage(errMsg)
-    setAlertSeverity(status)
-    alertRef.current?.triggerAlert()
-  }
+      let updatedData = params.data
 
-  const {
-    data: clonedFuelCodeData,
-    isLoading: isFuelCodeLoading,
-    refetch
-  } = useQuery({
-    queryKey: ['fuelCode', cloneFuelCodeId, prefix],
-    queryFn: async ({ queryKey }) => {
-      // eslint-disable-next-line no-unused-vars
-      const [_, cloneFuelCodeId, prefix] = queryKey
-      let path = apiRoutes.fuelCodeSearch
-      path +=
-        'prefix=' +
-        (prefix || 'BCLCF') +
-        '&distinctSearch=false&fuelCode=' +
-        cloneFuelCodeId
-      const response = await apiService.get(path)
-      return response.data.fuelCodes[0]
-    },
-    enabled: !!cloneFuelCodeId
-  })
-  const onCellValueChanged = useCallback(
-    (params) => {
-      if (!isEqual(params.oldValue, params.newValue)) {
-        params.data.modified = true
+      if (params.data.fuelCode !== undefined && params.data.fuelCode) {
+        const fuelCodeData = optionsData.latestFuelCodes.find(
+          (fuelCode) =>
+            fuelCode.fuelCode.split('.')[0] ===
+            params.data.fuelCode.split('.')[0]
+        )
+
+        updatedData = {
+          ...updatedData,
+          prefix: fuelCodeData.fuelCodePrefix.prefix,
+          company: fuelCodeData.company,
+          fuel: fuelCodeData.fuelCodeType.fuelType,
+          feedstock: fuelCodeData.feedstock,
+          feedstockLocation: fuelCodeData.feedstockLocation,
+          feedstockMisc: fuelCodeData.feedstockMisc,
+          feedstockTransportMode: fuelCodeData.feedstockFuelTransportModes.map(
+            (mode) => mode.feedstockFuelTransportMode.transportMode
+          ),
+          finishedFuelTransportMode:
+            fuelCodeData.finishedFuelTransportModes.map(
+              (mode) => mode.finishedFuelTransportMode.transportMode
+            ),
+          formerCompany: fuelCodeData.formerCompany,
+          contactName: fuelCodeData.contactName,
+          contactEmail: fuelCodeData.contactEmail
+        }
       }
-      if (params.column.colId === 'fuelCode') {
-        setCloneFuelCodeId(params.newValue)
-        setPrefix(params.data.prefix)
-        setFocusedCell(params.column.colId)
-        setParamsData(params.data)
-        refetch()
-      } else if (params.column.colId === 'fuelProductionFacilityCity') {
-        const location = optionsData.fpLocations?.find(
-          (location) => location.fuelProductionFacilityCity === params.newValue
+      if (params.data.fuelProductionFacilityCity !== undefined) {
+        const location = optionsData.fpLocations.find(
+          (location) =>
+            location.fuelProductionFacilityCity ===
+            params.data.fuelProductionFacilityCity
         )
-        const updatedData = {
-          ...params.data,
+        updatedData = {
```

Comment:
I feel this logic needs to be moved to valueSetter inside schema, what do you prefer?

Reply:
Im not opposed to moving it to the valueSetter function.

My reasoning was to keep as much of the business logic in the parent component as possible instead of splitting it up as its sometimes harder to find or map out whats happening.

I think normally how AG Grid does it is that they have the coldefs in the same file as the parent component as a state, making it easy to centralize the business logic. So i guess because we are splitting up the coldefs to a different file, it feels like we are splitting the business logic.

It doesnt look like this block of logic has any dependencies other than the `optionsData` so it would be an easy move. Lets do that.